### DISCUSSION TITLE
Using AddImplication vs Add().OnlyEnforceIf vs AddBoolOr

### DISCUSSION THREAD

**Discussion by ngaihiang** [google/or-tools#3216] [CP-SAT questions] [answered]
Using AddImplication vs Add().OnlyEnforceIf vs AddBoolOr

a = model.NewBoolVar('a') 
b = model.NewBoolVar('b')

If I have a case when a is true, then b is true. 

Is there merit to choosing one over the other?

- model.AddImplication(a, b)
- model.Add(b==1).OnlyEnforceIf(a)
- model.AddBoolOr(a.Not(), b)

Thanks in advance!

**Comment by lperron** [comment, marked as answer]
Add implication directly creates the addboolor. 
The second one is the worse in term of creation/presolve time. 

They will all create the same clause in the sat solver.
